ecl/hql/hqlgram2.cpp

```diff
@@ -8899,9 +8899,17 @@ bool HqlGram::checkRecordCreateTransform(HqlExprArray & assigns, IHqlExpression
             }
 
             if (rightExpr->isDatarow())
-                return checkRecordCreateTransform(assigns, leftRecord, leftSelected, rightRecord, rightSelected, errPos);
+            {
+                HqlExprArray childAssigns;
+                OwnedHqlExpr childSelf = getSelf(leftRecord);
+                if (!checkRecordCreateTransform(childAssigns, leftRecord, childSelf, rightRecord, rightSelected, errPos))
+                    return false;
+
+                IHqlExpression * transform = createValue(no_transform, makeTransformType(LINK(leftRecord->queryType())), childAssigns);
+                rightSelected.setown(createRow(no_createrow, transform));
+            }
 
-            assigns.append(*createAssign(LINK(leftSelected), checkEnsureRecordsMatch(leftSelected, rightSelected, errPos, false)));
+            assigns.append(*createAssign(LINK(leftSelected), checkEnsureRecordsMatch(leftSelected, rightSelected, errPos, rightExpr->queryType()->getTypeCode())));
```

Comment:
@ghalliday why isn't the 4th parameter rightSelected->queryType()->getTypeCode()?

Reply:
Changed